packages/cardpay-sdk/sdk/utils.ts

```diff
@@ -62,3 +117,178 @@ export function waitForEvent(contract: Contract, eventName: string, opts: PastEv
+      SafeTx: [
+        { type: 'address', name: 'to' },
+        { type: 'uint256', name: 'value' },
+        { type: 'bytes', name: 'data' },
+        { type: 'uint8', name: 'operation' },
+        { type: 'uint256', name: 'safeTxGas' },
+        { type: 'uint256', name: 'baseGas' },
+        { type: 'uint256', name: 'gasPrice' },
+        { type: 'address', name: 'gasToken' },
+        { type: 'address', name: 'refundReceiver' },
+        { type: 'uint256', name: 'nonce' },
+      ],
+    },
+    domain: {
+      verifyingContract: gnosisSafeAddress,
+    },
+    primaryType: 'SafeTx',
+    message: {
+      to: to,
+      value: value,
+      data: data,
+      operation: operation,
+      safeTxGas: txGasEstimate,
+      baseGas: baseGasEstimate,
+      gasPrice: gasPrice,
+      gasToken: txGasToken,
+      refundReceiver: refundReceiver,
+      nonce: nonce.toNumber(),
+    },
+  };
+  const signatureBytes = [];
+  signatureBytes.push(await signTypedData(web3, owner, typedData));
+
+  return signatureBytes;
+}
+
+export async function executeTransaction(
+  web3: Web3,
+  from: string,
+  to: string,
+  value: number,
+  data: any,
+  operation: number,
+  safeTxGas: string,
+  dataGas: string,
+  gasPrice: string,
+  nonce: string,
+  signatures: any,
+  gasToken: string,
+  refundReceiver: string
+): Promise<GnosisExecTx> {
+  let relayServiceURL = await getConstant('relayServiceURL', web3);
+  const url = `${relayServiceURL}/v1/safes/${from}/transactions/`;
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json', //eslint-disable-line @typescript-eslint/naming-convention
+    },
+    body: JSON.stringify({
+      to,
+      value,
+      data,
+      operation,
+      safeTxGas,
+      baseGas: dataGas,
+      dataGas,
+      gasPrice,
+      nonce,
+      signatures,
+      gasToken,
+      refundReceiver,
+    }),
+  };
+  let response = await fetch(url, options);
+  if (!response?.ok) {
+    throw new Error(await response.text());
+  }
+  return response.json();
+}
+
+async function signTypedData(web3: Web3, account: string, data: any): Promise<Signature> {
+  let result: string = await new Promise((resolve, reject) => {
+    let provider = web3.currentProvider;
+    if (typeof provider === 'string') {
+      throw new Error(`The provider ${web3.currentProvider} is not supported`);
+    }
+    if (provider == null) {
+      throw new Error('No provider configured');
+    }
+    //@ts-ignore TS is complaining that provider might be undefined--but the
+    //check above should prevent that from ever happening
+    provider.send(
+      {
+        jsonrpc: '2.0',
+        method: 'eth_signTypedData',
+        params: [account, data],
+        id: new Date().getTime(),
+      },
+      (err, response) => {
+        if (err) {
+          return reject(err);
+        }
+        resolve(response?.result);
+      }
+    );
+  });
+  const sig = result.replace('0x', '');
```

Comment:
@habdelra OK if we extract this RSV conversion to a separate util method, so signTypedData returns the hex string? For the auth stuff, I'd like to send to the server as the hex string.

Reply:
Totally fine by me